### PR TITLE
Clarifying that .map() returns None if None.

### LINKED ISSUE #107622: Make it clear that .map() returns None if None.
### Location

https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.map

### Summary

I think it would be nice if the .map() documentation made it clear that when None is consumed by .map() returns None, to make it consistent with .map_or() and .map_or_else() because right now it just says: 

> Maps an `Option<T> `to `Option<U>` by applying a function to a contained value.

Example in .map.or()

> "Returns the provided default result (if none), or applies a function to the contained value (if any)."

https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.map_or.

and maybe add a little example like:

```
let x: Option<&str> = None;
assert_eq!(x.map(|s| s.len()), None);
```

### PR DESCRIPTION
**Pull request by ameknite**
Fix #107622

### CODE REVIEW

**Comment by ameknite** [comment]
r+ @jyn514

File: library/core/src/option.rs
Comment (jyn514) [inline library/core/src/option.rs:946]:
```suggestion
    /// Maps an `Option<T>` to `Option<U>` by applying a function to a contained value (if `Some`) or returns `None` (if `None`).
```

**Review by jyn514 — APPROVED** [review]
Thanks! This looks good to me with the nit fixed :)

**Comment by jyn514** [comment]
@ameknite please don't push changes to a PR after been approved, it confuses GitHub when bors merges a commit that's not the same as the one in the PR.

@bors r- r+

**Pull request merged**
